src/util/triplea/MapXMLCreator/CanalDefinitionsPanel.java

```diff
@@ -0,0 +1,499 @@
+package util.triplea.MapXMLCreator;
+
+import games.strategy.util.Tuple;
+
+import java.awt.Color;
+import java.awt.FontMetrics;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.Point;
+import java.awt.event.ActionEvent;
+import java.awt.event.MouseEvent;
+import java.awt.geom.Rectangle2D;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+import javax.swing.AbstractAction;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+
+import util.triplea.MapXMLCreator.TerritoryDefinitionDialog.DEFINITION;
+
+/**
+ * 
+ * @author Erik von der Osten
+ * 
+ */
+public class CanalDefinitionsPanel extends ImageScrollPanePanel
+{
+	protected static final String NEW_CANAL_OPTION = "<new Canal>";
+
+	SortedSet<String> m_selectedLandTerritories = new TreeSet<String>();

```

Comment:
> "m_" naming prefixes: It helps reading the code and it used widely in the project. Why force breaking with it now? How other way to easily recognize by the name that a variable is a member variable?

I think the evidence is against "m_" as helping readability. That was a popular convention some time ago, as was hungarian notation and others. When it comes down to it, encoding extra information into variables is not that productive of a thing. It ties them to extra concepts, meanign if you want to simplyi make a variable static/non-static, then you have to change its name. Next, it's often not used consistently. Does it mean that whenever "m_" is missing that it is a method argument? How about the cases where we saw "m_" added in with method arg names. In general though its also a barrier to entry as well, why bother making people learn yet another convention when there are way more important things to keep track of. Which gets to readability, if you can't tell a variable is method local by looking at the method (on one page), then the method is not sized well. Same thing for class variables, ie: if you *have* to rely on "m_"  to know the scope of a variable, you're going to run into a ton more way more serious complexity problems, it's just a giant smell that complexity is running out of control. Lastly though, the team agreed some time ago to drop the convention when we moved to Github.

> SortedSet/TreeSet vs Set/HashSet: Reason was that I do not want the order of the territories to be changed in the XML file. So I need to be able to keep the same territory sorting I read from the canal definition in the XML.

Why not List + ArrayList?